### PR TITLE
#14 Added zoom to second screen

### PR DESCRIPTION
**Pull request by NikitaSukhanov**
- Added move and scale otptios for nonogram
- Rewrited some 2nd screen ui classes

### CODE REVIEW

File: app/src/main/java/com/mithridat/nonoconverter/ui/result/ResultActivity.java [hunk 1]
```diff
@@ -31,13 +30,13 @@ protected void onCreate(Bundle savedInstanceState) {
...
+        NonogramView nonorgamView = findViewById(R.id.nonogram_view);
         Field field =
                 getIntent()
                         .getParcelableExtra(ActivitiesConstants.EX_NONO_FIELD);
-        if (nonogramDrawer != null) {
-            nonogramMarginSet(nonogramDrawer);
-            nonogramDrawer.setNonogramField(field);
+        if (nonorgamView != null) {
+            nonorgamView.setNonogramField(field);
+            nonorgamView.setOnTouchListener(nonorgamView);
```
Comment (maik-nack) [inline app/src/main/java/com/mithridat/nonoconverter/ui/result/ResultActivity.java]: You can call this method in the constructor of the `NonogramView`:
```java
        setOnTouchListener(this);
```

File: app/src/main/java/com/mithridat/nonoconverter/ui/result/NonogramView.java [hunk 2]
```diff
@@ -0,0 +1,434 @@
...
+     */
+    private float _centerRelativeX = 0f;
+    private float _centerRelativeY = 0f;
+
+    /**
+     * Flag, marking if current state is restored.
+     */
+    private boolean _restoreState = false;
+
+
```
Comment (maik-nack) [inline app/src/main/java/com/mithridat/nonoconverter/ui/result/NonogramView.java]: Optional line

File: app/src/main/java/com/mithridat/nonoconverter/ui/result/NonogramView.java [hunk 3]
```diff
@@ -0,0 +1,434 @@
...
+        bundle.putFloat(StringKeys.CENTER_RELATIVE_X, _centerRelativeX);
+        bundle.putFloat(StringKeys.CENTER_RELATIVE_Y, _centerRelativeY);
+        return bundle;
+    }
+
+    @Override
+    protected void onRestoreInstanceState(Parcelable state) {
+        if (state instanceof Bundle) {
+            Bundle bundle = (Bundle) state;
+            super.onRestoreInstanceState(bundle.getParcelable(StringKeys.INITIAL_STATE));
```
Comment (maik-nack) [inline app/src/main/java/com/mithridat/nonoconverter/ui/result/NonogramView.java]: The length of this line over than 80.

File: app/src/main/java/com/mithridat/nonoconverter/ui/result/NonogramView.java [hunk 4]
```diff
@@ -0,0 +1,434 @@
...
+     * and then removes itself.
+     */
+    private void setListener() {
+        final View thisView = this;
+        final ViewTreeObserver observer = this.getViewTreeObserver();
+        ViewTreeObserver.OnGlobalLayoutListener listener =
+                new ViewTreeObserver.OnGlobalLayoutListener() {
+                    @Override
+                    public void onGlobalLayout() {
+                        if (_restoreState) restorePosition();
```
Comment (maik-nack) [inline app/src/main/java/com/mithridat/nonoconverter/ui/result/NonogramView.java]: Where are `{}`?

File: app/src/main/java/com/mithridat/nonoconverter/ui/result/NonogramView.java [hunk 5]
```diff
@@ -0,0 +1,434 @@
...
+
+    /**
+     * Class for touch-processing.
+     */
+    private class TouchManager {
+        /**
+         * Maximal and minimal total scale values.
+         */
+        static final float TOTAL_SCALE_MAX = 10f;
+        static final float TOTAL_SCALE_MIN = 0.8f;
```
Comment (maik-nack) [inline app/src/main/java/com/mithridat/nonoconverter/ui/result/NonogramView.java]: You can make these constants `private`.

File: app/src/main/java/com/mithridat/nonoconverter/ui/result/NonogramView.java [hunk 6]
```diff
@@ -0,0 +1,434 @@
...
+        }
+
+        /**
+         * Set 'scale' as current total scale at the moment.
+         * Used to restore view state.
+         *
+         * @param scale - scale to set
+         */
+        void setTotalScale(float scale) {
+            if (scale > TOTAL_SCALE_MAX)
```
Comment (maik-nack) [inline app/src/main/java/com/mithridat/nonoconverter/ui/result/NonogramView.java]: Where are `{}`?

File: app/src/main/java/com/mithridat/nonoconverter/ui/result/NonogramView.java [hunk 7]
```diff
@@ -0,0 +1,434 @@
...
+         * Scale detector.
+         */
+        private ScaleGestureDetector _scaleDetector;
+
+        /**
+         * Coordinates of the last touch.
+         */
+        private float _lastTouchX = 0f, _lastTouchY = 0f;
+
+
```
Comment (maik-nack) [inline app/src/main/java/com/mithridat/nonoconverter/ui/result/NonogramView.java]: Optional line

File: app/src/main/java/com/mithridat/nonoconverter/ui/result/NonogramView.java [hunk 8]
```diff
@@ -0,0 +1,434 @@
...
+                /* The last finger is removed from the display */
+                case MotionEvent.ACTION_UP:
+                case MotionEvent.ACTION_CANCEL:
+                    _activePointerId = INVALID_POINTER_ID;
+                    break;
+            }
+        }
+
+
```
Comment (maik-nack) [inline app/src/main/java/com/mithridat/nonoconverter/ui/result/NonogramView.java]: Optional line.

File: app/src/main/java/com/mithridat/nonoconverter/ui/result/NonogramView.java [hunk 9]
```diff
@@ -0,0 +1,434 @@
...
+                    }
+                    break;
+                }
+
+                /* The last finger is removed from the display */
+                case MotionEvent.ACTION_UP:
+                case MotionEvent.ACTION_CANCEL:
+                    _activePointerId = INVALID_POINTER_ID;
+                    break;
+            }
```
Comment (maik-nack) [inline app/src/main/java/com/mithridat/nonoconverter/ui/result/NonogramView.java:408]: Please add `default`.

File: app/src/main/java/com/mithridat/nonoconverter/ui/result/NonogramView.java [hunk 10]
```diff
@@ -0,0 +1,442 @@
...
+        if (centerDistanceX < -nonoWidth / 2f)
+            _nonogramPos.x += centerDistanceX + nonoWidth / 2f;
+        if (centerDistanceY > nonoHeight / 2f)
+            _nonogramPos.y += centerDistanceY - nonoHeight / 2f;
+        if (centerDistanceY < -nonoHeight / 2f)
+            _nonogramPos.y += centerDistanceY + nonoHeight / 2f;
+
+        nonoCenterX = _nonogramPos.x + nonoWidth / 2f;
+        nonoCenterY = _nonogramPos.y + nonoHeight / 2f;
+        _centerRelativeX = (nonoCenterX - viewCenterX) / nonoWidth;
```
Comment (maik-nack) [inline app/src/main/java/com/mithridat/nonoconverter/ui/result/NonogramView.java:277]: Why you don't use `centerDistanceX` here?

File: app/src/main/java/com/mithridat/nonoconverter/ui/result/NonogramView.java [hunk 11]
```diff
@@ -0,0 +1,442 @@
...
+        _centerRelativeY = (nonoCenterY - viewCenterY) / nonoHeight;
```
Comment (maik-nack) [inline app/src/main/java/com/mithridat/nonoconverter/ui/result/NonogramView.java:278]: Why you don't use `centerDistanceY` here?

**Review by maik-nack — APPROVED** [review]
IT IS WONDERFUL! EXCELLENT!

**Review by MaximSmolskiy — APPROVED** [review]
Look's great!

**Pull request merged**